hermes-common/src/main/java/pl/allegro/tech/hermes/common/kafka/KafkaConsumerPool.java

```diff
@@ -0,0 +1,78 @@
+package pl.allegro.tech.hermes.common.kafka;
+
+import com.google.common.cache.*;
+import java.util.Properties;
+import kafka.common.TopicAndPartition;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import pl.allegro.tech.hermes.common.broker.BrokerDetails;
+import pl.allegro.tech.hermes.common.broker.BrokerStorage;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.kafka.clients.consumer.ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.GROUP_ID_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.RECEIVE_BUFFER_CONFIG;
+
+public class KafkaConsumerPool {
```

Comment:
Virtually we can cope without this pool and create consumers on demand of retransmission but I think that this conception is not bad at all.

Reply:
I agree with you that we can use this pool. However it would be nice to add some comment above this class explaining why we are using it as it is not so obvious.